src/httpify/client_test.go

```diff
@@ -99,25 +98,49 @@ func TestClient_do(t *testing.T) {
 
 	server.Close()
 
+	var request uint64
+	var mut sync.Mutex
+	server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		mut.Lock()
+		if request == 0 {
+			request++
+			mut.Unlock()
+			time.Sleep(time.Second)
+		} else {
+			mut.Unlock()
+		}
+	}))
+
+	client, _ = NewClient(Params{
+		Domain:  server.URL,
+		Timeout: time.Millisecond,
+	})
+	client.baseURL.Scheme = "http"
+
+	assert.NotNil(t, client)
+	assert.Nil(t, err)
+
+	_, err = client.do("POST", "/assets.json", body, nil)
+	assert.Nil(t, err)
+	server.Close()
+
 	server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		time.Sleep(3 * time.Second)
 	}))
-	defer server.Close()
 
 	client, _ = NewClient(Params{
-		Domain:   server.URL,
-		Timeout:  time.Nanosecond,
-		APILimit: time.Nanosecond,
+		Domain:  server.URL,
+		Timeout: time.Millisecond,
 	})
+	client.maxRetry = 1
 	client.baseURL.Scheme = "http"
 
 	assert.NotNil(t, client)
 	assert.Nil(t, err)
 
 	_, err = client.do("POST", "/assets.json", body, nil)
-	if assert.NotNil(t, err) {
-		assert.Equal(t, err, errClientTimeout)
-	}
+	assert.EqualError(t, err, "request timed out after 1 retries, there may be an issue with your connection")
```

Comment:
Will the end user see this message? I can't tell where it gets outputted (and should it have a newline?).

Reply:
Yes the user will see this message if the request times out (5) times, the error is built with fmt.Errorf

Reply:
https://github.com/Shopify/themekit/pull/747/files#diff-2d68343eeed320d6dcac7d933fba1f00R127